### DISCUSSION TITLE
Can Viper convert between formats?

### DISCUSSION THREAD

**Discussion by flibby-jibbit** [spf13/viper#1799] [Q&A]
Suppose I have a configuration file in TOML, can I read that in and then write it out in YAML?

**Comment by hexvalid** [comment]
Yes. Read the TOML file and write to a filename with a YAML extension:

```go
func convertConfig() error {
    v := viper.New()
    v.SetConfigFile("config.toml")
    if err := v.ReadInConfig(); err != nil {
        return err
    }
    return v.SafeWriteConfigAs("config.yaml")
}
```

Import `github.com/spf13/viper`. `SafeWriteConfigAs` returns an error if the destination already exists; use `WriteConfigAs` when you intentionally want to overwrite it.

I tested this with Viper v1.21.0 and read the generated YAML back into a fresh instance: nested string and integer values were preserved. The output encoder is chosen from the destination extension in [`writeConfig`](https://github.com/spf13/viper/blob/v1.21.0/viper.go).

This converts configuration values; comments, original formatting, and key casing are not preserved. Use a fresh instance for the conversion so unrelated defaults or environment/flag overrides are not included in the output.
